### DISCUSSION TITLE
Adding Back Gless 2 Renderer for Godot 4

### DISCUSSION THREAD

**Discussion by ahmutkadim** [godotengine/godot-proposals#7057] [Engine Core]
I do not know whether renderer can be considered for engine core but I guess it kinda is. I would propose bringing back gless 2 renderer for Godot 4 since I think some platforms are not ready for gless 2. I believe apple still does not support gless 3 renderer which causes my web gemes to be unaccessible in all iOS devices and Safari browser on MacOS. I know that it would take a lot of work to implement and it is mostly apple s problem. However apple is not known for taking immidiate action for technological advancements and will not be interested in this topic since Unity web games work on IOS. They are popular ones so apple would not care if gless 3 is not supported on their devices. I would not like to develop for specifically IOS since it is hard for publishing in Apple App Store. I create web gems so that everyone can play and gless 2 renderer would really enhance accessibility of my game.  I know gless 2 renderer is limited but such limitations are part of web game development so I do not believe it would be a problem for any of developers. I cannot use Godot 3 either since Godot 4 has very good and hard to give up features like gdscript 2.0, better OS integration and text font setup. So please bring gless 2 renderer back as an option for compatibility renderer for Godot 4.

**Comment by Zireael07** [comment]
Already asked, and IIRC by you?

Also: if you want to develop only for the web, you're probably better off using something solely web-based (Phaser, GDevelop, heck, even roll something of your own with JS+WebGL). I found that the hard way. Godot projects on web take ages to load/to first interaction.

**Comment by AThousandShips** [comment]
Please format your post in a more readable way, this is one giant block of text that's impossible to read

Second, this isn't an issue with GLES, it's about WebGL2, as iOS *only* supports GLES 3 and not GLES 2, but you've already been informed about this
* godotengine/godot#77795

So please read the details over there again

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot/issues/77795.
